assets/dev/js/utils/plugin-actions.js

```diff
@@ -0,0 +1,259 @@
+		return response;
+	},
+
+	async install( url ) {
+		const installResponse = await this.asyncFetch( url );
+		return installResponse;
+	},
+
+	async activate( eventData ) {
+		if ( ! eventData ) {
+			return;
+		}
+
+		if ( eventData?.loaderTab ) {
+			this.showLoaderTab( eventData?.installText ?? eventData?.loaderTitle );
+		}
+		const activateResponse = await this.asyncFetch( eventData.url.replaceAll( '&amp;', '&' ) );
+		const activateBody = await activateResponse.text();
+		if ( this.requestHasErrors( activateBody ) ) {
+			return this.handleError( 'Plugin activation failed. Please try again.' );
+		}
+		this.handleFollowLinkAndRefresh( eventData );
+	},
+
+	async installAndActivate( eventData ) {
+		if ( ! eventData ) {
+			return;
+		}
+		if ( eventData?.loaderTab ) {
+			this.showLoaderTab( eventData?.installText ?? eventData?.loaderTitle );
+		}
+
+		if ( eventData?.preInstallCampaign ) {
+			this.handlePreInstallCampaign( eventData?.preInstallCampaign );
+		}
+
+		const installResponse = await this.asyncFetch( eventData.url.replaceAll( '&amp;', '&' ) );
+		const installBody = await installResponse.text();
+		if ( this.requestHasErrors( installBody ) ) {
+			return this.handleError( 'Plugin installation failed. Please try again.' );
+		}
+		this.updateLoaderHeading( eventData?.activateText ?? eventData?.loaderTitle );
+		const activateLink = this.getActivateLink( installBody, eventData );
+		if ( false === activateLink ) {
+			this.handleError( 'Plugin activation link not found. Please try again.' );
+		}
+		const activateResponse = await this.asyncFetch( activateLink.replaceAll( '&amp;', '&' ) );
+		const activateBody = await activateResponse.text();
+		if ( this.requestHasErrors( activateBody ) ) {
+			return this.handleError( 'Plugin activation failed. Please try again.' );
+		}
+
+		this.handleFollowLinkAndRefresh( eventData );
+	},
+
+	handlePreInstallCampaign( campaign ) {
+		if ( campaign?.action ) {
+			elementorCommon.ajax.addRequest( campaign.action, { data: { ...campaign.data } } );
+		}
+	},
+
+	handleError( errorMessage ) {
+		this.hideLoaderTab();
+		// eslint-disable-next-line no-alert
+		return alert( errorMessage );
+	},
+
+	handleFollowLinkAndRefresh( eventData ) {
+		if ( eventData?.followLink ) {
+			this.loaderTab.location.href = eventData?.followLink;
+			this.loaderTab.focus();
+		} else {
+			this.hideLoaderTab();
+		}
+
+		if ( eventData?.refresh ) {
+			window.location.reload();
+		}
+	},
+
+	requestHasErrors( body ) {
+		return body.indexOf( 'error-page' ) > -1;
+	},
+
+	/**
+	 * Open a new tab with a loading animation.
+	 * @param {string} heading
+	 */
+	showLoaderTab( heading = 'Loading...' ) {
+		const winHtml = `<!DOCTYPE html>
+<html>
+    <head></head>
+    <body>
+        <style>
+			#elementor-loading {
+			    position: fixed;
+			    inset: 0;
+			    background: #fff;
+			    z-index: 9999;
+			}
+			.elementor-loader-wrapper {
+			    position: absolute;
+			    top: 50%;
+			    left: 50%;
```

Comment:
Can you replace this with `inset-*` please, for logical properties support. And bellow too.